x-pack/packages/ml/data_view_utils/components/create_data_view_switch.tsx

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { type FC } from 'react';
+
+import { EuiSwitch } from '@elastic/eui';
+
+import { i18n } from '@kbn/i18n';
+
+interface CreateDataViewSwitchProps {
+  canCreateDataView: boolean;
+  createDataView: boolean;
+  setCreateDataView: (d: boolean) => void;
+}
+
+export const CreateDataViewSwitch: FC<CreateDataViewSwitchProps> = ({
+  canCreateDataView,
+  createDataView,
+  setCreateDataView,
+}) => (
+  <EuiSwitch
```

Comment:
You need to add the 'Create data view' switch into the end of the advanced JSON editor for DFA jobs - this is how it looks on `main` currently:

<img width="526" alt="image" src="https://github.com/elastic/kibana/assets/7405507/07e2266b-0582-4a03-95fa-b4f0396dd158">